### ISSUE TITLE
go/internal/generate returns weird values when build errors

### ISSUE THREAD

**Issue by gucio321** [golang/go#46207]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.16.4 linux/amd64

</pre>

### Does this issue reproduce with the latest release?

yes

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/user/.cache/go-build"
GOENV="/home/user/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/user/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/user/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.16.4"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/dev/null"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build3225948580=/tmp/go-build -gno-record-gcc-switches"

</pre></details>

### What did you do?

Hi!
I was changing my project's structure. In my `enums` directory I have files with `//go:generate stringer` directives. At some point, I forgot to change package name in a file copied from somewhere else. I've tried to run `go generate` and and got stuck:
```console
stringer: no value for constant type2s1
file2.go:3: running "stringer": exit status 1
```

<details><summary>example</summary>

there are two files in package, but with another `package` names
the `go build .` returns:
```console
found packages package1 (file1.go) and package2 (file2.go) in /home/user/test/
```

but `go generate` returns something strange:
```console
stringer: no value for constant type2s1
file2.go:3: running "stringer": exit status 1
```

<details><summary>file1.go</summary>

```golang
package package1

//go:generate stringer -linecomment -type type1

type type1 byte

const (
	type1s1 type1 = iota
	type1s2
)
```

</details>

<details><summary>file2.go</summary>

```golang
package package2

//go:generate stringer -linecomment -type type2

type type2 byte

const (
	type2s1 type2 = iota
	type2s2
)
```

</details>

</details> 

### What did you expect to see?

`go generate` should report understandable message when build errors found

### What did you see instead?
```console
stringer: no value for constant type2s1
file2.go:3: running "stringer": exit status 1

```

**Comment by dmitshur** [comment]
`go generate` invokes commands. If a command fails, `go generate` display the output from said command. A command may fail because there are build errors, or it might fail for another completely unrelated reason.

So, if there's anything to do here, it's to improve `stringer` to print better error messages when it fails due to a build error. I'll retitle the issue to be about that.

**Comment by dmitshur** [comment]
CC @mvdan via [owners](https://dev.golang.org/owners).

**Comment by gucio321** [comment]
>go generate invokes commands. If a command fails, go generate displays the output from said command. A command may fail because there are build errors, or it might fail for another completely unrelated reason.

good point. Just couldn't found reproducable example with calling `stringer`

**Comment by gucio321** [comment]
oh, okey, so:
in my example calling `stringer -type type2 .` will crash as described
```console
stringer: no value for constant type2s1
```
